modules/gltf/extensions/physics/gltf_document_extension_physics.cpp

```diff
@@ -202,71 +352,87 @@ GLTFMeshIndex _get_or_insert_mesh_in_state(Ref<GLTFState> p_state, Ref<ImporterM
+		p_state->add_used_extension("OMI_physics_shape");
 	}
-	Array state_colliders;
-	if (omi_collider_ext.has("colliders")) {
-		state_colliders = omi_collider_ext["colliders"];
+	Array state_shapes;
+	if (omi_physics_shape_ext.has("shapes")) {
+		state_shapes = omi_physics_shape_ext["shapes"];
 	} else {
-		omi_collider_ext["colliders"] = state_colliders;
+		omi_physics_shape_ext["shapes"] = state_shapes;
+	}
+	return state_shapes;
+}
+
+Dictionary _export_node_shape(Ref<GLTFState> p_state, Ref<GLTFPhysicsShape> p_physics_shape) {
+	Array state_shapes = _get_or_create_state_shapes_in_state(p_state);
+	int size = state_shapes.size();
+	Dictionary shape_property;
+	Dictionary shape_dict = p_physics_shape->to_dictionary();
+	for (int i = 0; i < size; i++) {
+		Dictionary other = state_shapes[i];
+		if (other == shape_dict) {
+			// De-duplication: If we already have an identical shape,
+			// set the shape index to the existing one and return.
+			shape_property["shape"] = i;
+			return shape_property;
+		}
 	}
-	return state_colliders;
+	// If we don't have an identical shape, add it to the array.
+	state_shapes.push_back(shape_dict);
+	shape_property["shape"] = size;
+	return shape_property;
 }
 
 Error GLTFDocumentExtensionPhysics::export_node(Ref<GLTFState> p_state, Ref<GLTFNode> p_gltf_node, Dictionary &r_node_json, Node *p_node) {
-	Dictionary node_extensions = r_node_json["extensions"];
+	Dictionary physics_body_ext;
 	Ref<GLTFPhysicsBody> physics_body = p_gltf_node->get_additional_data(StringName("GLTFPhysicsBody"));
 	if (physics_body.is_valid()) {
-		node_extensions["OMI_physics_body"] = physics_body->to_dictionary();
-		p_state->add_used_extension("OMI_physics_body");
+		physics_body_ext = physics_body->to_dictionary();
 	}
-	Ref<GLTFPhysicsShape> collider = p_gltf_node->get_additional_data(StringName("GLTFPhysicsShape"));
-	if (collider.is_valid()) {
-		Array state_colliders = _get_or_create_state_colliders_in_state(p_state);
-		int size = state_colliders.size();
-		Dictionary omi_collider_ext;
-		node_extensions["OMI_collider"] = omi_collider_ext;
-		Dictionary collider_dict = collider->to_dictionary();
-		for (int i = 0; i < size; i++) {
-			Dictionary other = state_colliders[i];
-			if (other == collider_dict) {
-				// De-duplication: If we already have an identical collider,
-				// set the collider index to the existing one and return.
-				omi_collider_ext["collider"] = i;
-				return OK;
-			}
-		}
-		// If we don't have an identical collider, add it to the array.
-		state_colliders.push_back(collider_dict);
-		omi_collider_ext["collider"] = size;
+	Ref<GLTFPhysicsShape> collider_shape = p_gltf_node->get_additional_data(StringName("GLTFPhysicsColliderShape"));
+	if (collider_shape.is_valid()) {
+		physics_body_ext["collider"] = _export_node_shape(p_state, collider_shape);
+	}
+	Ref<GLTFPhysicsShape> trigger_shape = p_gltf_node->get_additional_data(StringName("GLTFPhysicsTriggerShape"));
```

Comment:
Nitpick but I think to make compile-time StringName, you're supposed to use `SNAME`

Reply:
Please see the comment above the SNAME macro "If in doubt, do not use":

```cpp
/*
 * The SNAME macro is used to speed up StringName creation, as it allows caching it after the first usage in a very efficient way.
 * It should NOT be used everywhere, but instead in places where high performance is required and the creation of a StringName
 * can be costly. Places where it should be used are:
 * - Control::get_theme_*(<name> and Window::get_theme_*(<name> functions.
 * - emit_signal(<name>,..) function
 * - call_deferred(<name>,..) function
 * - Comparisons to a StringName in overridden _set and _get methods.
 *
 * Use in places that can be called hundreds of times per frame (or more) is recommended, but this situation is very rare. If in doubt, do not use.
 */
```